### PR TITLE
Lims 229 adding more columns

### PR DESCRIPTION
**Pull request by DanE417**
## Description of the issue/feature this PR addresses

Linked issue: https://bika.atlassian.net/browse/LIMS-229, https://bika.atlassian.net/browse/LIMS-820, https://bika.atlassian.net/browse/LIMS-821

## Current behavior before PR
Extra columns were required to be added to Analysis Services, Analysis Profiles, Sample Templates, Clients, Client Contacts, Instruments.

## Desired behavior after PR is merged
Columns were added to Analysis Services, Analysis Profiles, Sample Templates, Clients, Client Contacts, Instruments.

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

**Comment by luthando5** [comment]
Hey @DanE417 looking very good! Are you planning on making a new branch for each listing change or is this PR not ready for review yet?

File: src/bika/extras/browser/listingview/samples.py [hunk 1]
```diff
@@ -44,4 +47,13 @@ def folder_item(self, obj, item, index):
...
             item["Container"] = container_title
             item["replace"]["Container"] = container_link
+        
+        specification = full_object.getSpecification()
+        if specification:
+            spec_title = specification.Title()
+            spec_url = specification.absolute_url()
+            spec_link = get_link(spec_url, spec_title)
+            item["Specification"] = spec_title
+            item["replace"]["Specification"] =spec_link
```
Comment (luthando5) [inline src/bika/extras/browser/listingview/samples.py:57]: So far so good :+1:

File: src/bika/extras/browser/overrides/overrides.zcml
Comment (luthando5) [inline src/bika/extras/browser/overrides/overrides.zcml:1]: Hopefully there will be a fix for this before this PR is ready for review.

**Comment by DanE417** [comment]
This does not include LIMS-822 and LIMS-823

File: src/bika/extras/browser/listingview/analysisservices.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            self.listing.review_states[i]["columns"].append("Description")
+            self.listing.review_states[i]["columns"].append("CommercialID")
+            self.listing.review_states[i]["columns"].append("DecimalPrecision")
+            self.listing.review_states[i]["columns"].append("ProtocolID")
+            self.listing.review_states[i]["columns"].append("Price")
+            self.listing.review_states[i]["columns"].append("Vat")
+            self.listing.review_states[i]["columns"].append("BulkPrice")
+            self.listing.review_states[i]["columns"].append("Hidden")
+
+    def folder_item(self, obj, item, index):
```
Comment (luthando5) [inline src/bika/extras/browser/listingview/analysisservices.py:81]: We use the replace when we are adding a link to the value. Please remove them if not adding links.

File: src/bika/extras/browser/listingview/bika_analysisprofiles.py [hunk 3]
```diff
@@ -56,9 +67,24 @@ def folder_item(self, obj, item, index):
...
+        item["CommercialID"] = commercialID
+
+        # VAT
+        vat = obj.getAnalysisProfileVAT()
+        item["Vat"] = vat
+
+        # Use Analysis Profile Price
+        use_price = obj.getUseAnalysisProfilePrice()
+        use_price_value = _("Yes") if use_price else _("No")
+        item["replace"]["UsePrice"] = use_price_value
```
Comment (luthando5) [inline src/bika/extras/browser/listingview/bika_analysisprofiles.py]: Remove the replace please :)

File: src/bika/extras/browser/listingview/sampletemplates.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            sample_type_title = sample_type.Title()
+            sample_type_url = sample_type.absolute_url()
+            sample_type_link = get_link(sample_type_url, sample_type_title)
+            item["SampleTypeTitle"] = sample_type_title
+            item["replace"]["SampleTypeTitle    "] = sample_type_link
+
+        # Composite
+        composite = obj.getComposite()
+        composite_value = _("Yes") if composite else _("No")
+        item["replace"]["Composite"] = composite_value
```
Comment (luthando5) [inline src/bika/extras/browser/listingview/sampletemplates.py]: Remove replace please.

**Review by luthando5 — CHANGES_REQUESTED** [review]

File: src/bika/extras/browser/listingview/sampletemplates.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            item["replace"]["SamplePointTitle"] = sample_point_link
+
+        # Sample Type
+        sample_type = obj.getSampleType()
+        if sample_type:
+            sample_type_title = sample_type.Title()
+            sample_type_url = sample_type.absolute_url()
+            sample_type_link = get_link(sample_type_url, sample_type_title)
+            item["SampleTypeTitle"] = sample_type_title
+            item["replace"]["SampleTypeTitle    "] = sample_type_link
```
Comment (luthando5) [inline src/bika/extras/browser/listingview/sampletemplates.py]: Remove extra spaces please.

**Comment by luthando5** [comment]
Well done on your first PR @DanE417 :1st_place_medal:

**Review by luthando5 — APPROVED** [review]

**Pull request merged**